connectors/migrations/20241211_fix_gdrive_parents.ts

```diff
@@ -60,19 +59,24 @@ async function migrate({
           id: {
             [Op.gt]: nextId,
           },
+          mimeType: "application/vnd.google-apps.folder",
         },
         order: [["id", "ASC"]],
         limit: QUERY_BATCH_SIZE,
       }
     );
 
-    for (const file of googleDriveFiles) {
+    loop: for (const file of googleDriveFiles) {
       const internalId = file.dustFileId;
       const driveFileId = file.driveFileId;
       const parents = [driveFileId];
       let current: string | null = file.parentId;
       while (current) {
         parents.push(current);
+        if (typeof parentsMap[current] === "undefined") {
```

Comment:
aren't we guaranteed to err at some point? since at some point we get to the root drive, which is not in googleDriveFiles right? 
YPATH so I'm likely missing something there :sweat_smile:

Reply:
the root dirve is indeed in googleDriveFiles, so in most of the case we don't err :-) 
 But I found many places where we have parentId which do not match any google_drive_file (in that case I can't guess the parents list)